### DISCUSSION TITLE
Is there a way to specify one out of a set of combinations of fields are required?

### DISCUSSION THREAD

**Discussion by cafeclimber** [pydantic/pydantic#3602] [Question] [answered]
For example:

```python
from pydantic import BaseModel

class MyModel(BaseModel):
    a: str
    b: Optional[str] = None
    c: Optional[str] = None
    d: Optional[str] = None
```

In `MyModel` is there a pythonic or pydantic way to ensure either `(b, c)`, `(c, d)`, or `(b, d)` are defined?

**Comment by samuelcolvin** [comment, marked as answer]
Best way is with a `validator` or `root_valudator`.

Or you could create three different models, one with `b` one with `c` one with `d`, then use a union.
